### PR TITLE
Make the header and footer patterns text visible on all themes

### LINKED ISSUE #35248: [GlobalStep] Unable to view "Text and Icon" for dark mode on Templates page.
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Unable to view "Text and Icon" for dark mode on Templates page.

### Expected behavior

"Text and Icon "should be displayed for dark mode on Templates page.

### Actual behavior

Unable to view "Text and Icon" for dark mode on Templates page.

### Steps to reproduce

1. Create any test site using Jurassic Ninja.
2. Install and activate all the required plugins.
3. In a new page or a template, go to the patterns tab.
4. Make sure all 6 dark patterns for header and footer appear under the WooCommerce category dropdown.
4. Select each dark mode header and footer pattern.
5. Observe that, we are unable to view "Text and Icon" for dark mode on Templates page.

### Instruction link:
https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/testing/releases/870.md#add-minimal-large-and-essential-header-patterns-7292
https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/testing/releases/870.md#add-simple-large--two-menus-footer-patterns-7306

### Screenshot:
Header:
minimal dark
essential dark
large dark

Footer:
simple footer dark
2 menus footer dark
Large footer dak

### WordPress Environment

WooCommerce  7.1 beta 1
WordPress version: 6.1-beta2

PC:
Windows 10
Chrome(Version 106.0.5249.119)
Firefox(Version 106.0.1)

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

<details>

`
### WordPress Environment ###

WC Version: 7.1.0
REST API Version: ✔ 7.1.0
WC Blocks Version: ✔ 8.7.1
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.1-RC2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.54 (Unix) OpenSSL/1.0.2g
PHP Version: 7.4.32
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.33-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.1.0
WC Database Prefix: wp_
Total Database Size: 6.63MB
Database Data Size: 5.13MB
Database Index Size: 1.50MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 4.05MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.11MB + Index: 0.06MB + Engine InnoDB
wp_posts: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 24
page: 8
post: 2
product: 18
product_variation: 7
revision: 1
wp_global_styles: 2
wp_template_part: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (4) ###

Query Monitor: by John Blackbourn – 3.10.1
Companion Plugin: by Osk – 1.28
WooCommerce: by Automattic – 7.1.0-beta.1
WordPress Beta Tester: by Peter Westwood
Andy Fragen – 3.2.3

### Inactive Plugins (3) ###

Akismet Anti-Spam: by Automattic – 5.0.1
Hello Dolly: by Matt Mullenweg – 1.7.2
Jetpack: by Automattic – 11.4

### Dropin Plugins (1) ###

db.php: Query Monitor Database Class (Drop-in)

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/
Checkout: #9 - /checkout/
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Three
Version: 1.0
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-10-21 10:45:45 +00:00
Options: ✔
Notes: 36
Onboarding: skipped

### Action Scheduler ###

Complete: 36
Oldest: 2022-10-20 10:48:45 +0000
Newest: 2022-10-21 06:25:54 +0000

Pending: 1
Oldest: 2022-10-21 10:47:46 +0000
Newest: 2022-10-21 10:47:46 +0000

### Status report information ###

Generated at: 2022-10-21 10:41:36 +00:00
`

</details>

### PR DESCRIPTION
**Pull request by albarin**
The goal of this PR is to make the text visible on all themes on the WooCommerce header/footer patterns.
The light patterns will inherit styles from the theme and the dark ones will have black background and white text.

Fixes https://github.com/woocommerce/woocommerce/issues/35248
#### User Facing Testing

1. Create a new page and insert all the WooCommerce Header and Footer patterns (dark & light, 12 patterns in total)
2. Save, go to the frontend and check the text and icons are visible.
3. Change the theme to TT3 and check again. Test with a few more different TT3 theme styles.

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Fix the text color for the header and footer WooCommerce patterns.

### CODE REVIEW

**Review by Aljullu — COMMENTED** [review]
I wonder if a better approach would be to remove all colors from the patterns and let them "inherit" the ones from the theme. That will ensure the patterns work in all themes, with any color, and maybe it would allow us to only have one variant of them instead of having dark/light ones.

Personally, I think that's a better approach than having one with white background and another one with black background. Specially because with TT3 many style variations have peculiar backgrounds like images and gradients so white/black headers/footers might not work well.

Do you think that's doable, @albarin?

**Comment by albarin** [comment]
> I wonder if a better approach would be to remove all colors from the patterns and let them "inherit" the ones from the theme. That will ensure the patterns work in all themes, with any color, and maybe it would allow us to only have one variant of them instead of having dark/light ones.
> 
> Personally, I think that's a better approach than having one with white background and another one with black background. Specially because with TT3 many style variations have peculiar backgrounds like images and gradients so white/black headers/footers might not work well.
> 
> Do you think that's doable, @albarin?

@Aljullu Yes, I agree, I've been testing the approach you mentioned and it works ok with multiple themes and style variations.
Should we then remove all colors and just have one pattern variant for each header and footer (meaning no dark & light versions)?
Pinging @vivialice and @nerrad to see also what they think about it.

Examples:
Screenshot 2022-10-31 at 09 45 05
Screenshot 2022-10-31 at 09 48 01
Screenshot 2022-10-31 at 09 48 56
Screenshot 2022-10-31 at 09 52 32

**Comment by nerrad** [comment]
why not both gif

Seriously though, I think we should have both unopinionated (inherit styles mostly from the theme) and opinionated patterns. So I think it's okay to have some dark header/footer patterns which require more opinionated styling. 

I'm also interested in VIv's feedback here.

**Comment by vivialice** [comment]
👍 The main/light patterns should take style direction from the theme. I don't know if there's a way to determine that for dark versions, but I think we can define the color there otherwise.

**Comment by albarin** [comment]
Thanks all for your feedback.
I've made the light patterns inherit all the styles possible from the theme. Then for dark patterns, I've set the background to black and text/icons to white. It seems to work well for all the themes and variations I've tried.
Could you give it another try @Aljullu? Thanks!

**Review by Aljullu — COMMENTED** [review]
Thanks for updating this PR, @albarin!

>  I've made the light patterns inherit all the styles possible from the theme. Then for dark patterns, I've set the background to black and text/icons to white. It seems to work well for all the themes and variations I've tried.

Sounds like a good solution. I would however remove the `Light` word from light patterns name and just have the "regular one" and the "dark one". Ie:
* WooCommerce Large Footer
* WooCommerce Large Footer | Dark

---

I found a couple of small issues, mentioning them below: :point_down: 

* Some lights patterns have hardcoded color for the social networks icons. I think we should remove it even if that means diverging from the original designs. Otherwise, there is no way to make them look good in all backgrounds:

imatge

* In _WooCommmerce Essential Header | Light_ and _WooCommerce Large Header | Light_, links are still dark:

imatge

imatge

**Comment by albarin** [comment]
@Aljullu thanks for the review, just pushed fixes to the 3 issues 🙂

File: patterns/footer-large.php [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
- * Slug: woocommerce-blocks/footer-large-light
+ * Title: WooCommerce Large Footer
+ * Slug: woocommerce-blocks/footer-large
  * Categories: WooCommerce
  */
 ?>
 
-<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"32px","right":"4%","bottom":"32px","left":"4%"},"blockGap":"40px"},"elements":{"link":{"color":{"text":"var:preset|color|black"}}}},"backgroundColor":"background","textColor":"black"} -->
-<div class="wp-block-group alignfull has-black-color has-background-background-color has-text-color has-background has-link-color" style="padding-top:32px;padding-right:4%;padding-bottom:32px;padding-left:4%"><!-- wp:columns {"align":"wide","style":{"spacing":{"blockGap":"16px"}}} -->
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"32px","right":"4%","bottom":"32px","left":"4%"},"blockGap":"40px"}},"backgroundColor":"background"} -->
```
Comment (Aljullu) [inline patterns/footer-large.php]: I didn't test, but the issue might be caused by the `backgroundColor` defined here?

**Review by Aljullu — COMMENTED** [review]
Thanks for updating the PR, @albarin!

I could find another case where the patterns don't look good. I tried with Twenty Twenty Two modifying the colors (making the background black and the text color light):

imatge

then, the light WC patterns don't look good:

imatge

File: patterns/footer-large.php (hunk 1 above)
Comment (albarin) [reply patterns/footer-large.php]: Thanks for the thorough testing, that was the issue 😅  

Sorry for all the back and forth, hopefully that fixes it 🙏

File: patterns/footer-large.php [hunk 2]
```diff
@@ -29,11 +29,15 @@
 <!-- /wp:column -->
 
 <!-- wp:column {"width":"20%","style":{"spacing":{"padding":{"top":"0px"}}}} -->
-<div class="wp-block-column" style="padding-top:0px;flex-basis:20%"><!-- wp:navigation {"layout":{"type":"flex","orientation":"vertical"}} /--></div>
+<div class="wp-block-column" style="padding-top:0px;flex-basis:20%"><!-- wp:navigation {"layout":{"type":"flex","orientation":"vertical"}} -->
+<!-- wp:page-list /-->
+<!-- /wp:navigation --></div>
```
Comment (Aljullu) [inline patterns/footer-large.php]: I think this should go back to not have the `wp:page-list` block:
```suggestion
<div class="wp-block-column" style="padding-top:0px;flex-basis:20%"><!-- wp:navigation {"layout":{"type":"flex","orientation":"vertical"}} /--></div>
```

(There are several other instances of this in other light patterns as well)

**Review by Aljullu — APPROVED** [review]
Style-wise it looks perfect! :ok_hand: I noticed one last issue, however: menus seem to be different in the light and dark patterns:

imatge

I think that's only reproducible if you have created at least one menu (modifying the _Navigation_ block). I left a comment inline where I think the source of the issue might be.

I'm pre-approving, but happy to re-review if you want. :slightly_smiling_face:

**Comment by vivialice** [comment]
👋 Looking good! I am wondering if in your screenshot @Aljullu - the site name is in black? Making the text look not quite centred?

**Comment by Aljullu** [comment]
> Looking good! I am wondering if in your screenshot @Aljullu - the site name is in black? Making the text look not quite centred?

The site name is `latest`, it appears after the year. Text doesn't appear centered in my screenshot because of the way I took it (cutting off the right side :sweat_smile: ). This is a full-page screenshot in the frontend:

imatge

**Pull request merged**